### DISCUSSION TITLE
v3.0.0

### DISCUSSION THREAD

**Discussion by ALTaleX531** [ALTaleX531/TranslucentFlyouts#111] [Announcements]
## New Registry Keys and Items
```
HKEY_CURRENT_USER\Software\TranslucentFlyouts\BlockList
HKEY_CURRENT_USER\Software\TranslucentFlyouts\DisabledList
HKEY_CURRENT_USER\Software\TranslucentFlyouts\EnableMiniDump (dword)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\DarkMode_ThemeColorizationType (string)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\LightMode_ThemeColorizationType (string)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\DisabledList
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\EnableCompatibilityMode (dword)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\DarkMode_ThemeColorizationType (string)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\LightMode_ThemeColorizationType (string)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\DropDown\Animation
HKEY_CURRENT_USER\Software\TranslucentFlyouts\DropDown\DisabledList
HKEY_CURRENT_USER\Software\TranslucentFlyouts\DropDown\EnableFluentAnimation (dword)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\DisabledList
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\DarkMode_Color (dword)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\LightMode_Color (dword)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\MarginsType (dword)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\Margins_cxLeftWidth (dword)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\Margins_cxRightWidth (dword)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\Margins_cyTopHeight (dword)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\Margins_cyBottomHeight (dword)
```
I don't have time to document each of these new features and registry items due to academic reasons, but you can ask what they do in `Discussion` and I'll respond later.
This version fixes almost all the bugs, please try it.

## New Contributors
* @ddzzkunye made their first contribution in https://github.com/ALTaleX531/TranslucentFlyouts/pull/67

**Full Changelog**: https://github.com/ALTaleX531/TranslucentFlyouts/compare/v2.1.0...v3.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ALTaleX531/TranslucentFlyouts/releases/tag/v3.0.0'>v3.0.0</a>.</em>

**Comment by Satanarious** [comment]
Does it also fix the problem where it somehow messes with vanguard, the anti-cheat in Valorant?

**Reply by ALTaleX531** [reply]
In my tests I can start Valorant without problems, they seem to have changed the anti-cheat algorithm, you can confirm this yourself.

**Comment by Satanarious** [comment]
If you get time, I would appreciate it if you documented the changes. Otherwise, I would require the following information to support TF v3.0:
- Values acceptable for all new keys
- What purpose does each key serve and how any acceptable value change the appearance or behaviour of TF

Though I tested nothing, I made the following speculations: 
- I see you have BlockList and DisabledList, which I suppose accept process names as keys and 0 or 1 as DWORD values to signify blocked/disabled status. Let me know more about this.
- There is now an Animation key for DropDown. We currently have animation in the menu section in TFC. Is it similar to this or something else?
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/9d830871-6efd-4d3c-b22b-16897efcddb1)
- We already have DarkModeBorderColor and DarkModeGradientColor and their light mode counterparts, what is DarkModeColor for in Tooltip? You only used DarkModeColor and LightModeColor in TF/Menu keys. Is this being used in favor of removing the former two?
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/4650010c-7ffd-44a5-a927-85dbf41d7a54)


Let me know if I need to remove anything from the previous build of TFC, like keys or something. And don't forget the first thing I asked.

**Reply by ALTaleX531** [reply]
```
HKEY_CURRENT_USER\Software\TranslucentFlyouts\BlockList
HKEY_CURRENT_USER\Software\TranslucentFlyouts\DisabledList
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\DisabledList
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\DisabledList
HKEY_CURRENT_USER\Software\TranslucentFlyouts\DropDown\DisabledList
```
As you can see, they do store a list of process names, but the values have slightly different meanings, here's a succinct explanation.
<img width="632" alt="image" src="https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/7ef95865-4ec2-489e-9f37-057cfecf4760">
However, please note that BlockList should only be used for special purposes and should not be set by general users.

DropDown's `Animation` is the same as Menu's, there's no difference.

```
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\DarkMode_Color
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\LightMode_Color
```
These are for defining the tooltip text colors, as they have been hardcoded as 0xFFFFFFFF and 0xFF1A1A1A respectively in previous versions. Note that the Alpha channel has no effect.

```
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\MarginsType
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\Margins_cxLeftWidth
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\Margins_cxRightWidth
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\Margins_cyTopHeight
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\Margins_cyBottomHeight
```
These items are used to define the margins of the tooltips. In V3, their default value is 6.
<img width="840" alt="image" src="https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/38123adc-8709-4cd3-af35-2eaaa9d92496">
MarginsType defines how the following four items are used in the calculations. 0 is stacked directly on top of the default margins, and 1 replaces the default margins. here is a code snippet to help understand.
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/9d5c1184-221c-4ef7-8ff6-ac12ecabe153)

```
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\DarkMode_ThemeColorizationType (REG_MULTI_SZ)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\LightMode_ThemeColorizationType (REG_MULTI_SZ)
...
```
Remember `EnableThemeColorization`? Now the above item comes with it together.
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/c78d0046-5a89-4690-8f5d-dfd5b414ef98)
In previous versions, they were hardcoded as `ImmersiveStartHoverBackground`. All possible names can be found at [list-immersive-colors](https://github.com/pbatard/list-immersive-colors/blob/master/list-immersive-colors.txt).

`HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\EnableCompatibilityMode`. One of its main features is that it no longer downloads the uxtheme symbol file.  Some of the rendering logic falls back to v1. This is useful if you don't want to download symbol files. Not downloading symbol files at all also requires disabling `NoModernAppBackgroundColor`

`HKEY_CURRENT_USER\Software\TranslucentFlyouts\EnableMiniDump` is enabled by default. When it is disabled, TF can no longer generate dump files, which is often used in special scenarios. Note that explorer.exe always has minidump enabled, even if this option is disabled, for security reasons.

**Reply by Satanarious** [reply]
I think I understood everything with just a couple of doubts. What is the difference between BlockList and DisabledList? I understand that BlockList is not meant for general user but for the application itself and for special cases. What about DisabledList in that context?

Also, There are too many options for DarkModeThemeColorizationType, do you have any suggestions on how I should display those?

 Edit: What is the use of anything in BlockList or DisabledList if it has a value of 0?

**Reply by ALTaleX531** [reply]
Processes in the BlockList are prevented from loading TFMain64.dll, but if the process had loaded TFMain64.dll before it was added to the BlockList, it will still reside in the process memory. Why is it not recommended to use BlockList? Because there are programs that will keep trying to load TFMain64.dll until they succeed, which will use too much CPU.
For processes in the DisabledList, TF disables most of the functionality, TFMain64.dll still resides in the process memory and it can dynamically respond to registry changes.


```
ImmersiveStartBackground
ImmersiveStartHoverBackground
ImmersiveSystemAccent
ImmersiveSystemAccentDark1
ImmersiveSystemAccentDark2
ImmersiveSystemAccentDark3
ImmersiveSystemAccentLight1
ImmersiveSystemAccentLight2
ImmersiveSystemAccentLight3
```
I think the above types are sufficient, and many of them behave exactly the same as they do.

**Reply by Satanarious** [reply]
I get it now. Thanks for the info. A few people are a little too displeased with TFC not supporting TF v3.0 the same day, so I guess I'll start working on the next update soon.

**Reply by Satanarious** [reply]
You didn't answer my question about process names in BlockedList and DisabledList with value of 0.

**Reply by ALTaleX531** [reply]
As I said before, items with a value of 0 are treated as if they were not included in the list, as if they did not exist.

**Reply by Satanarious** [reply]
Got it.

**Reply by ALTaleX531** [reply]
>
> ```
> HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\DarkMode_ThemeColorizationType (REG_MULTI_SZ)
> HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\LightMode_ThemeColorizationType (REG_MULTI_SZ)
> ...
> ```
Correction: it's REG_SZ not REG_MULTI_SZ.

**Reply by Satanarious** [reply]
I would also like to know the default value for each key.

Edit: By the above I mean, what keys need to be present to be considered and what default values are assumed in the absence of those keys?

**Reply by Satanarious** [reply]
```
HKEY_CURRENT_USER\Software\TranslucentFlyouts\DropDown\DarkMode_ThemeColorizationType (string)
HKEY_CURRENT_USER\Software\TranslucentFlyouts\DropDown\LightMode_ThemeColorizationType (string)
```

Do these exist for DropDown? Or you forget to mention them? Because there is an `EnableThemeColorization` key for DropDown.

**Reply by ALTaleX531** [reply]
All keys exist with corresponding default values.
`EnableMiniDump` -> `1`
`DarkMode_ThemeColorizationType` | `LightMode_ThemeColorizationType` -> `ImmersiveStartHoverBackground`
`HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\DarkMode_Color` -> `0xFFFFFFFF`
`HKEY_CURRENT_USER\Software\TranslucentFlyouts\Tooltip\LightMode_Color` -> `0xFF1A1A1A`
`MarginsType` -> `0`
`Margins_cxLeftWidth` | `Margins_cxRightWidth` | `Margins_cyTopHeight` | `Margins_cyBottomHeight`   -> `6`

In fact, DarkMode_ThemeColorizationType and LightMode_ThemeColorizationType exist everywhere EnableThemeColorization exists, so I've omitted them to avoid taking up too much space.

**Comment by e-t-l** [comment]
Between releases v3.0.0 and v3.0.1, it seems like you sometimes include 
TranslucentFlyouts.Win32.V3.x86.optional.rar  with TFmain32.dll, and sometimes not. What determines whether or not the 32-bit version is included?

**Reply by ALTaleX531** [reply]
The x86 version wasn't ready when v3.0.0 was released, so it wasn't included.

**Comment by A-Basir** [comment]
There is no install_win32.bat for the TranslucentFlyouts.Win32.V3.x86.optional but there is a launch_win32.bat for that
But I have to manually open it after each reboot ( for now I added launch_win32.bat to startup )
Did you forget to add it or is there a reason for that? like compatibility issues?

**Reply by ALTaleX531** [reply]
I think I forgot to add it, thanks for the reminder. You can change the parameter of the command inside `launch_win32.bat` to `/install` to install it.

**Reply by A-Basir** [reply]
I checked the `launch_win32.bat` and actually the command is `/install` but somehow I need to start it every reboot

**Comment by Satanarious** [comment]
I am sorry to bother you again, but I forgot if you need a restart after altering the DisabledList or BlockList. Can you help?

Here's a peek into the new DisabledList UI 😁,

![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/3f4d9f06-dff2-40bd-aae0-98307e0eb6d8)

**Reply by ALTaleX531** [reply]
DisabledList takes effect immediately after the settings are changed, while BlockList requires them to restart for apps added later.
<img width="212" alt="image" src="https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/0e37d95b-c2a8-4a7e-b932-7df8f3bfb6d9">
I think the button to enable/disable process items also needs to be added, other than that this UI has been a great surprise to me.

**Reply by Satanarious** [reply]
Up until now, I removed any value that had a default value. `0` for any item in `DisabledList` means that the value has no effect even if it is present in the key. But knowing that it takes effect immediately, a switch for enable/disable seems logical. I will make the required additions. 

Thanks for the feedback.

**Comment by Satanarious** [comment]
<table>
<tr>
<img src="https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/d35a9ef0-1c96-44a1-b374-890920a1b5ae" width=50%>
<img src="https://github.com/ALTaleX531/TranslucentFlyouts/assets/42642682/3dd05301-4b50-4704-8a7e-4d5b27505e56" width=50%>
</tr>
</table>


These are the colors for each mentioned type, right? I'm doubtful because I don't know what these values do.

**Reply by ALTaleX531** [reply]
These colors are not fixed, they will adjust slightly with your theme colors so it is unlikely to preview them. These colors are often used in different parts of the UI, with darker colors indicating less important background surfaces. Important surfaces are highlighted with brighter and lighter colors. To be honest, these values come from disassembling Microsoft's system components, and I know no more about it than you do.

**Reply by Satanarious** [reply]
I think I understand now. So, should I not show the colors at all, and show only the values? How would the user understand the values though?

**Reply by ALTaleX531** [reply]
Of course, it's not completely impossible to preview, it's just that I think it's hard to implement on the python side. At the moment it looks like you can only show the values and tell the user that these are variants of the color, after all, almost no one knows what these values are actually used for.

**Comment by Satanarious** [comment]
I'm writing descriptions for the new values. I would like some more details on:
```
- DarkModeThemeColorizationType
- LightModeThemeColorizationType
- EnableCompatibilityMode
```

**Reply by ALTaleX531** [reply]
`DarkMode_ThemeColorizationType` and `LightMode_ThemeColorizationType` are used to indicate which variant of the current theme color to use, which are usually darker and lighter, and are used in various parts of the UI.
`EnableCompatibilityMode` will enable a v1-like rendering procedure for your vanilla menus, so symbols are no longer needed for that section, but the rendering will be performed slightly more slowly.
